app/models/manageiq/providers/embedded_ansible/automation_manager/playbook_runner.rb

```diff
@@ -130,11 +130,28 @@ def queue_signal(*args, deliver_on: nil)
     )
   end
 
-  def playbook
-    ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(options[:playbook_id])
+  def temp_configuration_script
+    ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript.new(
+      :manager     => playbook.manager,
+      :manager_ref => options[:job_template_ref],
+      :variables   => {}
+    )
   end
 
-  def tower_job
-    ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job.find(options[:tower_job_id])
+  def delete_inventory
+    return unless options[:inventory]
+    playbook.manager.with_provider_connection do |connection|
```

Comment:
It seems like this `inventory.destroy!` logic should be moved into the `manageiq-providers-ansible_tower` gem here: `/app/models/manageiq/providers/ansible_tower/shared/automation_manager/tower_api.rb` To keep all the calls to `with_provider_connection` consistent.

Reply:
Yes. I agree if the `Inventory` class included `TowerApi`. 
@bdunne Any reason https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/automation_manager/inventory.rb#L1 (or its parent class) does not include `TowerApi`?

Reply:
I think `TowerApi` is a newer module and didn't get included everywhere yet.

Reply:
I tried to include `TowerApi` but it did not work. `Inventory` is a subclass of `EmsFolder` which fails to retrieve the provider object.
We can attempt to include `TowerApi` with another PR if it is doable.